stdlib/LinearAlgebra/test/cholesky.jl

```diff
@@ -399,4 +404,25 @@ end
     @test B ≈ Matrix(C)
 end
 
+@testset "CholeskyPivoted and Factorization" begin
+    A = randn(8,8)
+    B = A'A
+    C = cholesky(B, Val(true), check=false)
+    @test CholeskyPivoted{eltype(C)}(C) === C
+    @test Factorization{eltype(C)}(C) === C
+    @test Array(CholeskyPivoted{complex(eltype(C))}(C)) == Array(cholesky(complex(B), Val(true), check=false))
+    @test Factorization{complex(eltype(C))}(C) == cholesky(complex(B), Val(true), check=false)
+end
+
+@testset "REPL printing of CholeskyPivoted" begin
+    A = randn(8,8)
+    B = A'A
+    C = cholesky(B, Val(true), check=false)
+    cholstring = sprint((t, s) -> show(t, "text/plain", s), C)
+    rankstring = "$(C.uplo) factor with rank $(rank(C)):" 
+    factorstring = sprint((t, s) -> show(t, "text/plain", s), C.uplo == 'U' ? C.U : C.L)
+    permstring   = sprint((t, s) -> show(t, "text/plain", s), C.p)
+    @test cholstring == "$(summary(C))\n$rankstring\n$factorstring\npermutation:\n$permstring"
+end
```

Comment:
I'm not sure: do we test REPL printing usually? Seems like if somebody changes the printing, then (s)he is simply going to rewrite this whole test.

Well, I just saw that we do have a printing test in LU, so I guess this is fine then.

Reply:
We have it for SVD as well